apiserver/facades/client/usermanager/usermanager.go

```diff
@@ -383,3 +383,43 @@ func (api *UserManagerAPI) setPassword(arg params.EntityPassword) error {
 	}
 	return nil
 }
+
+// ResetPassword resets password for supplied users by
+// invalidating current passwords (if any) and generating
+// new random secret keys which will be returned.
+func (api *UserManagerAPI) ResetPassword(args params.Entities) (params.AddUserResults, error) {
+	var result params.AddUserResults
+
+	if err := api.check.ChangeAllowed(); err != nil {
+		return result, errors.Trace(err)
+	}
+
+	if len(args.Entities) == 0 {
+		return result, nil
+	}
+
+	isSuperUser, err := api.hasControllerAdminAccess()
```

Comment:
Agreed. So it sounds like at this stage, self-reset will not be of much value but might cause confusion.
Let's land this as is and either wait for users feedback/requests to reset their own password and addressed it with proper logout/re-authenticate mechanism.

For now, I think only controller admins being able to rest is sufficient... It might even be worthwhile to not allow controller admin to reset their own pwd... What do you think?